src/common/src/array/stream_chunk.rs

```diff
@@ -360,23 +360,9 @@ impl StreamChunk {
         }
     }
 
-    // Derive the chunk permits based on the provided rate limit
-    pub fn compute_rate_limit_chunk_permits(&self, limit: usize) -> usize {
-        let chunk_size = self.capacity();
-        let ends_with_update = if chunk_size >= 2 {
-            // Note we have to check if the 2nd last is `U-` to be consistenct with `StreamChunkBuilder`.
-            // If something inconsistent happens in the stream, we may not have `U+` after this `U-`.
-            self.ops()[chunk_size - 2].is_update_delete()
-        } else {
-            false
-        };
-        if chunk_size == limit + 1 && ends_with_update {
-            // If the chunk size exceed limit because of the last `Update` operation,
-            // we should minus 1 to make sure the permits consumed is within the limit (max burst).
-            chunk_size - 1
-        } else {
-            chunk_size
-        }
+    // Compute the required permits of this chunk for rate limiting.
+    pub fn compute_rate_limit_chunk_permits(&self) -> u64 {
+        self.capacity() as _
     }
```

Comment:
Can we remove it since there is no limit for burst?

Reply:
I slightly prefer keeping this, because if we don't have this, we may have difficulty to decide between `chunk.capacity()` and `chunk.cardinality()` later.